delivery-status-service/src/main/java/com/tracking_delivery_system/delivery_status_service/service/DeliveryStatusUpdater.java

```diff
@@ -0,0 +1,59 @@
+package com.tracking_delivery_system.delivery_status_service.service;
+
+import com.tracking_delivery_system.delivery_status_service.model.Deliver;
+import com.tracking_delivery_system.delivery_status_service.model.DeliveryStatus;
+import com.tracking_delivery_system.delivery_status_service.model.StatusUpdate;
+import com.tracking_delivery_system.delivery_status_service.repository.DeliverRepository;
+import lombok.RequiredArgsConstructor;
+import org.springframework.stereotype.Service;
+
+import java.util.LinkedHashMap;
+import java.util.Map;
+import java.util.UUID;
+import java.util.function.Predicate;
+
+@Service
+@RequiredArgsConstructor
+public class DeliveryStatusUpdater {
```

Comment:
Usually, in real projects class will newer be injected, and will be injected his interface instead. There are lot of reasons for that. So please create interface for this class, and every time you need this class just use his interface so you will avoid hard cohesion between services